code/modules/jobs/job_types/_job.dm

```diff
@@ -133,6 +133,10 @@
 	if(lock_flags || gimmick)
 		SSjob.job_manager_blacklisted |= title
 
+/// Called only once when someone joins as said job, AFTER equip() does it's thing. Override this instead of running some fancy snowflake code somewhere else.
+/// Latejoin is provided for the sheer convenience of having it.
+/datum/job/proc/initialize(mob/living/carbon/human/H, latejoin = FALSE)
```

Comment:
This code does not work at all. Equip() is called after all of the jobs are selected. To make what you are trying to do work requires a major overhaul to the job system which would be provided by #12643

Reply:
Also initialize is a completely misleading name for something that runs at the end of job selection.